app/src/main/java/org/dhis2/metadata/usecases/sdkextensions/SdkExtensions.kt

```diff
@@ -0,0 +1,258 @@
+package org.dhis2.metadata.usecases.sdkextensions
+
+import io.reactivex.Single
+import org.hisp.dhis.android.core.D2
+import org.hisp.dhis.android.core.arch.repositories.scope.RepositoryScope
+import org.hisp.dhis.android.core.category.CategoryOptionCombo
+import org.hisp.dhis.android.core.common.State
+import org.hisp.dhis.android.core.dataelement.DataElement
+import org.hisp.dhis.android.core.dataset.DataSetInstance
+import org.hisp.dhis.android.core.dataset.DataSetInstanceSummary
+import org.hisp.dhis.android.core.datavalue.DataValueConflict
+import org.hisp.dhis.android.core.enrollment.Enrollment
+import org.hisp.dhis.android.core.event.Event
+import org.hisp.dhis.android.core.imports.TrackerImportConflict
+import org.hisp.dhis.android.core.organisationunit.OrganisationUnit
+import org.hisp.dhis.android.core.period.Period
+import org.hisp.dhis.android.core.program.Program
+import org.hisp.dhis.android.core.program.ProgramStage
+import org.hisp.dhis.android.core.program.ProgramTrackedEntityAttribute
+import org.hisp.dhis.android.core.trackedentity.TrackedEntityAttribute
+import org.hisp.dhis.android.core.trackedentity.TrackedEntityInstance
+import org.hisp.dhis.android.core.trackedentity.TrackedEntityType
+import org.hisp.dhis.android.core.trackedentity.TrackedEntityTypeAttribute
+
+fun D2.programs(): List<Program> =
```

Comment:
This SdkExtensions is a good idea to abstract recurrent request into the SDK, why don't we move it to commons. Do you think it would make more sense?.
Could it be tested?